### PR TITLE
Lecture 10

### PR DESCRIPTION
**Pull request by alexandruradovici**
### Pull Request Overview

This pull request adds the slides for lecture 10.

### Testing Strategy

This pull request was tested by using the slides.

### TODO or Help Wanted

Feedback

### Build

- [x] Ran `npm build`.

### Author

Signed-off-by: Alexandru Radovici <alexandru.radovici@wyliodrin.com>

### CODE REVIEW

File: slides/lectures/10/os/slides.md [hunk 1]
```diff
@@ -0,0 +1,202 @@
...
+}
+</style>
+
+**Actions**
+- **process** and **threads**
+- use the *Processor* and *Accelerators* (GPU, Neural Engine, etc)
+
+**Data**
+- everything is a file
+- peripherals a viewed as files (*POSIX*)
```
Comment (DanutAldea) [inline slides/lectures/10/os/slides.md:62]:
```suggestion
- peripherals are viewed as files (*POSIX*)
```

File: slides/lectures/10/rtos/slides.md [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+- applications are similar to *threads* (are considered friendly)
+- the whole system is compiled into a single binary
+- similar to frameworks
+
+---
+---
+# Real Time?
+upper bound
+
+- **real time** means **performing** an action in **always** in a **deterministic** amount of **time**
```
Comment (DanutAldea) [inline slides/lectures/10/rtos/slides.md:30]:
```suggestion
- **real time** means **performing** an action **always** in a **deterministic** amount of **time**
```

File: slides/lectures/10/tock/slides.md [hunk 3]
```diff
@@ -0,0 +1,618 @@
...
+<div style="grid-column-start: 1; grid-column-end: 3;">
+
+- A **preemptive** embedded OS (runs on MCUs)
+  - Cortex-M
+  - RISC-V
+- Uses memory protection (**MPU required**)
+- Has separate **kernel and user space**
+  - most embedded OS have the one piece software philosophy
+- Runs untrusted apps in user space
+- **Microkernel** architecture
```
Comment (DanutAldea) [inline slides/lectures/10/tock/slides.md:31]:
```suggestion
- **Hybrid** architecture
```
Is Tock not a hybrid architecture?

File: slides/lectures/10/tock/slides.md [hunk 4]
```diff
@@ -0,0 +1,618 @@
...
+<img src="/tock/tock-layout-runtime.svg" class="w-80 rounded" />
+
+---
+---
+# Process States
+
+- Tock runs only on *single core*
+- *Running* state means the process is ready to run
+- *Yielded* means the process waits for an event (*upcall*)
+- *start* and *stop* are use commands
```
Comment (DanutAldea) [inline slides/lectures/10/tock/slides.md:177]:
```suggestion
- *start* and *stop* are user commands
```

File: slides/lectures/10/tock/slides.md [hunk 5]
```diff
@@ -0,0 +1,618 @@
...
+---
+# Faults
+similar to segfaults
+
+- the kernel and apps can fault
+- a detailed debug message can be displayed
+- due to MPU usage Tock apps fault on:
+  - trying to access memory outside its data (includes peripheral access)
+  - stack overflow
+  - trying to perform operations
```
Comment (DanutAldea) [inline slides/lectures/10/tock/slides.md:292]:
```suggestion
  - trying to perform privileged operations
```

**Review by DanutAldea — CHANGES_REQUESTED** [review]

**Review by ZamfirAndreiPaul — APPROVED** [review]

**Review by DanutAldea — APPROVED** [review]

**Review by ZamfirAndreiPaul — APPROVED** [review]

**Pull request merged**